### PR TITLE
feat!: fx and interscheme implementation

### PR DESCRIPTION
**Pull request by vijayg10**
**Changes:**
- Currency Conversion (FX)
  - FxQuotes API implementation in both simple routing mode and persistent mode
  - Adjusted participants validation to accommodate FX
- Interscheme
  - Proxy cache implementation using redis
  - Parameterize switch id
  - Adjusted participants validation to accommodate interscheme
- Performance optimisations
  - Option to send audit and trace events directly to Kafka instead of using sidecars
  - Allowed UUID v7 in the API patterns

### CODE REVIEW

**Review by geka-evk — APPROVED** [review]

File: docker/quoting-service/default.json [hunk 1]
```diff
@@ -45,11 +50,10 @@
     "includeCauseExtension": false,
     "truncateExtensions": true
   },
-  "SIMPLE_ROUTING_MODE": true,
+  "SIMPLE_ROUTING_MODE": false,
```
Comment (elnyry-sam-k) [inline docker/quoting-service/default.json:53]: Reason ?

**Comment by elnyry-sam-k** [comment]
A general comment: I couldn't find any changes to readme files corresponding to these updates and on the documentation repo, could only find the PR: https://github.com/mojaloop/documentation/pull/434 for the fx transfers sequences but not quotes.

Please update here if these are provided elsewhere or if they're still needed.

File: docker/quoting-service/default.json (hunk 1 above)
Comment (kalinkrustev) [reply docker/quoting-service/default.json:53]: @vijayg10 ^^

File: src/interface/QuotingService-swagger.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ info:
     - The API supports a maximum size of 65536 bytes (64 Kilobytes) in the HTTP header.
   license:
     name: Open API for FSP Interoperability (FSPIOP) (Implementation Friendly Version)
-  version: "1.1"
+  version: '1.1'
```
[the remark this answers is not in the document]
Comment (kalinkrustev) [reply src/interface/QuotingService-swagger.yaml:9]: @vijayg10 ^^

File: docker/quoting-service/default.json (hunk 1 above)
Comment (vijayg10) [reply docker/quoting-service/default.json:53]: This is for the integration tests and not the default value in config/default.json.
As part of this PR https://github.com/mojaloop/quoting-service/pull/346 integrations tests are hardened and need persistance in the quoting service.

File: src/interface/QuotingService-swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vijayg10) [reply src/interface/QuotingService-swagger.yaml:9]: Good catch. Yes it should be 2.0.

**Comment by elnyry-sam-k** [comment]
Thanks Steven, for the updates!

Few comments / questions for clarification:

1. Is there no ILP packet being returned in the `PUT /fxQuotes`? to confirm, is it just the `condition`?
2. There's also some` /services `related requests here, is the implementation for those provided here or is it just represented for context?

**Comment by oderayi** [comment]
> Thanks Steven, for the updates!
> 
> Few comments / questions for clarification:
> 
> 1. Is there no ILP packet being returned in the `PUT /fxQuotes`? to confirm, is it just the `condition`?
> 2. There's also some`/services`related requests here, is the implementation for those provided here or is it just represented for context?

1. Yes, we only need `condition`  in `PUT /fxQuotes`
2. I removed `services`. Vijay confirmed this is no longer implemented/supported but is added as a config in sdk-scheme-adapter.

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**